src/mds/MDLog.cc

```diff
@@ -651,13 +656,32 @@ void MDLog::trim(int m)
     ceph_assert(segments.size() >= pre_segments_size);
     max_expiring_segments = std::max<unsigned>(max_expiring_segments,segments.size() - pre_segments_size);
   }
-  
+
   map<uint64_t,LogSegment*>::iterator p = segments.begin();
+
+  auto trim_start = ceph::coarse_mono_clock::now();
+  std::optional<ceph::coarse_mono_time> trim_end;
+
+  auto log_trim_counter_start = log_trim_counter.get();
+  auto log_trim_threahold = g_conf().get_val<Option::size_t>("mds_log_trim_threshold");
+
   while (p != segments.end()) {
-    if (stop < ceph_clock_now())
-      break;
+    // throttle - break out of trimmming if we've hit the threahold
```

Comment:
```suggestion
    // throttle - break out of trimmming if we've hit the threahold
    dout(15) << __func__ << ": trim_counter is " << trim_counter << ": maybe trimming " << *p << dendl;
```

Or something like that.